pages/api/performances/collect.js

```diff
@@ -31,17 +31,19 @@ export default async (req, res) => {
     // We get all the performances where the competition_level, dance_size, dance_style are in
     // the requested settingIDs that are passed in
     const settingIDArray = settingIDs.split(',').map(i => +i);
-    filter.competition_level_id = {
-      in: settingIDArray,
-    };
 
-    filter.dance_size_id = {
-      in: settingIDArray,
-    };
-
-    filter.dance_style_id = {
-      in: settingIDArray,
-    };
+    for (const settingID of settingIDArray) {
+      const setting = await getSetting(settingID);
+      if (setting) {
+        if (setting.type === 'COMPETITION_LEVEL') {
+          filter.competition_level_id = settingID;
+        } else if (setting.type === 'STYLE') {
+          filter.dance_style_id = settingID;
+        } else if (setting.type === 'DANCE_SIZE') {
+          filter.dance_size_id = settingID;
+        }
+      }
+    }
   }
```

Comment:
This is an N+1 query problem. API performance shouldn't be too much worse but with the `Performances` page being the most important page, we should consider optimizing this.

Reply:
What could be done instead is that in awards/collect and awards/get, we can get the setting type along with the setting ids. Then, we wouldn't have to query to find the type in this for loop

Reply:
Decided to leave this out for now cc @mkanoria